mubench.reviewsite/src/RoutesHelper.php

```diff
@@ -19,24 +19,42 @@ class RoutesHelper
     private $logger;
     private $site_base_url;
     private $upload_path;
+    private $max_reviews;
 
-    public function __construct(DBConnection $db, PhpRenderer $renderer, Logger $logger, $upload_path, $site_base_url)
+    public function __construct(DBConnection $db, PhpRenderer $renderer, Logger $logger, $upload_path, $site_base_url, $max_reviews)
     {
         $this->db = $db;
         $this->renderer = $renderer;
         $this->logger = $logger;
         $this->site_base_url = $site_base_url;
         $this->upload_path = $upload_path;
+        $this->max_reviews = $max_reviews;
     }
 
     public function index(Request $request, Response $response, array $args) {
         return $this->render($this, $request, $response, $args, 'index.phtml', []);
     }
 
     public function detector(Request $request, Response $response, array $args) {
+        $max_reviews = $request->getQueryParam("max_reviews", $this->max_reviews);
         $detector = $this->db->getOrCreateDetector($args['detector']);
-        $runs = $this->db->getRuns($detector, $args['exp']);
-        return $this->render($this, $request, $response, $args, 'detector.phtml', ['runs' => $runs]);
+        $runs = [];
+        foreach($this->db->getRuns($detector, $args['exp']) as $run){
+            $index = 0;
```

Comment:
This is not really an index, but the number of misuses with a conclusive review state. Please rename.